src/main/java/com/minecolonies/coremod/client/gui/WindowHutAllInventory.java

```diff
@@ -0,0 +1,251 @@
+{
+
+    /**
+     * List of all item stacks in the warehouse.
+     */
+    List<ItemStorage> allItems = new ArrayList<>();
+
+    /**
+     * Resource scrolling list.
+     */
+    private final ScrollingList stackList;
+    /**
+     * The filter for the resource list.
+     */
+    private       String        filter         = "";
+    /**
+     * The sortDescriptor so how we want to sort
+     */
+    private       int           sortDescriptor = 0;
+
+    /**
+     * The building associated to the GUI.
+     */
+    private final IBuildingView building;
+
+    /**
+     * Constructor for a hut rename entry window.
+     *
+     * @param b {@link AbstractBuilding}
+     */
+    public WindowHutAllInventory(final IBuildingView b)
+    {
+        super(Constants.MOD_ID + HUT_ALL_INVENTORY_SUFFIX);
+        this.building = b;
+        registerButton(BUTTON_SORT, this::setSortFlag);
+        this.stackList = findPaneOfTypeByID(LIST_ALLINVENTORY, ScrollingList.class);
+        updateResources();
+    }
+
+    @Override
+    public void onOpened()
+    {
+
+    }
+
+    /**
+     * Increments the sortDescriptor and sets the GUI Button accordingly Valid Stages 0 - 4 NO_SORT         0   No Sorting, like wysiwyg ASC_SORT        1   Name Ascending
+     * DESC_SORT       2   Name Descending COUNT_ASC_SORT  3   Itemcount Ascending COUNT_DESC_SORT 4   Itemcount Descending
+     **/
+    private void setSortFlag()
+    {
+        sortDescriptor++;
+        if (sortDescriptor > 4)
+        {
+            sortDescriptor = NO_SORT;
+        }
+        switch (sortDescriptor)
+        {
+            case NO_SORT:
+                findPaneOfTypeByID(BUTTON_SORT, ButtonImage.class).setLabel("v^");
+                break;
+            case ASC_SORT:
+                findPaneOfTypeByID(BUTTON_SORT, ButtonImage.class).setLabel("A^");
+                break;
+            case DESC_SORT:
+                findPaneOfTypeByID(BUTTON_SORT, ButtonImage.class).setLabel("Av");
+                break;
+            case COUNT_ASC_SORT:
+                findPaneOfTypeByID(BUTTON_SORT, ButtonImage.class).setLabel("1^");
+                break;
+            case COUNT_DESC_SORT:
+                findPaneOfTypeByID(BUTTON_SORT, ButtonImage.class).setLabel("1v");
+                break;
+            default:
+                break;
+        }
+
+        updateResources();
+    }
+
+    /**
+     * Update the item list.
+     */
+    private void updateResources()
+    {
+        List<ItemStorage> filterItems = new ArrayList<>();
+        final List<BlockPos> containerList = building.getContainerList();
+        List<ItemStack> items = new ArrayList<>();
+        int count = containerList.size();
+        World world = building.getColony().getWorld();
+        for (int s = 0; s < count; s++)
+        {
+            final TileEntity rack = world.getTileEntity(containerList.get(s));
+            if (rack instanceof TileEntityRack)
```

Comment:
if not instanceof TileEntityRack but rack.getCapability(IItemHandlerCap) then  get itemhandler and iterate over slots.